### PR TITLE
Bug 1176650 - Refactored fetching client IPs.

### PR DESCRIPTION
**Pull request by jezdez**
This does a bunch of things:

- Introduce a middleware to use the guarenteed client IP from the load balancer via the X-Forwarded-For header.

- Combined the two existing tools that defer a unique identifier from the request.

- Cleaned up a few tests.

### CODE REVIEW

**Comment by jezdez** [comment]
@groovecoder Let me know when you review this, let's have a quick talk about it.

File: kuma/core/utils.py [hunk 1]
```diff
@@ -325,5 +288,36 @@ def chord_flow(pre_task, tasks, post_task):
 
 
 def limit_banned_ip_to_0(group, request):
-    ip = get_ip(request)
+    ip = request.META.get('REMOTE_ADDR', '10.0.0.1')
     return IPBanJob().get(ip)
+
+
+def get_unique(content_type, object_pk, name=None, request=None,
```
Comment (groovecoder) [inline kuma/core/utils.py]: :+1: consolidation!

**Comment by groovecoder** [comment]
code reviewed; about to spot-check

**Comment by groovecoder** [comment]
Spot-check looks good. I tampered an edit request and added `X-Forwarded-For` HTTP header, which then showed up in the revision dashboard when I toggled the IPs on.

**Pull request merged**